packages/brain/src/modules/cron/trackValidatorsPerformance.ts

```diff
@@ -0,0 +1,167 @@
+
+  const epochFinalized = await beaconchainApi.getEpochHeader({ blockId: "finalized" });
+  let newEpochFinalized = epochFinalized;
+
+  while (epochFinalized === newEpochFinalized) {
+    try {
+      logger.debug(`${logPrefix}Epoch finalized: ${epochFinalized}`);
+
+      // get validator indexes
+      // TODO: could be a better way to get the indexes? i.e store them in cache or brain DB
+      logger.debug(`${logPrefix}Getting validator indexes from pubkeys`);
+      const validatorIndexes = await Promise.all(
+        validatorPubkeys.map(
+          async (pubkey) => (await beaconchainApi.getStateValidator({ state: "finalized", pubkey })).data.index
+        )
+      );
+      logger.debug(`${logPrefix}Validator indexes: ${validatorIndexes}`);
+
+      // get only active validators
+      logger.debug(`${logPrefix}$Getting active validators`);
+      const activeValidators = (
+        await beaconchainApi.postStateValidators({
+          body: {
+            ids: validatorIndexes,
+            statuses: validatorIndexes.map(() => ValidatorStatus.ACTIVE_ONGOING)
+          },
+          stateId: "finalized"
+        })
+      ).data.map((validator) => validator.index);
+      logger.debug(`${logPrefix}Active validators: ${activeValidators}`);
+
+      if (activeValidators.length === 0) {
+        logger.info(`${logPrefix}No active validators found`);
+        return;
+      }
+
+      // check node health
+      logger.debug(`${logPrefix}Checking node health`);
+      const { el_offline, is_syncing } = (await beaconchainApi.getSyncingStatus()).data;
+      logger.debug(`${logPrefix}EL Node offline: ${el_offline}, Node syncing: ${is_syncing}`);
+      if (el_offline) throw Error("EL Node is offline");
+      if (is_syncing) throw Error("Node is syncing");
+
+      // get block attestations
+      logger.debug(`${logPrefix}Getting attestations rewards for epoch ${epochFinalized}`);
+      const validatorsAttestationsRewards = (
+        await beaconchainApi.getAttestationsRewards({
+          epoch: epochFinalized.toString(),
+          pubkeysOrIndexes: validatorIndexes
+        })
+      ).data.total_rewards;
+      logger.debug(`${logPrefix}Attestations rewards: ${JSON.stringify(validatorsAttestationsRewards)}`);
+
+      // insert performance data
+      for (const validatorIndex of validatorIndexes) {
+        //const liveness = validatorsLiveness.find((liveness) => liveness.index === validatorIndex)?.is_live;
+        const attestationsRewards = validatorsAttestationsRewards.find(
+          (liveness) => liveness.validator_index === validatorIndex
+        );
+
+        if (!attestationsRewards) {
+          logger.error(`${logPrefix}Missing data for validator ${validatorIndex}, att: ${attestationsRewards}`);
+          continue;
+        }
+
+        // write on db
+        logger.debug(`${logPrefix}Inserting performance data for validator ${validatorIndex}`);
+        await postgresClient.insertPerformanceData({
```

Comment:
Should we retry the whole loop if this insert fails or just give up for this epoch?